### ISSUE TITLE
[no-type-alias] unhandled type - shouldn't happen

### ISSUE THREAD

**Issue by j-o** [typescript-eslint/typescript-eslint#720]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-type-alias": ["error", { "allowAliases": "in-unions" }]
  }
}
```

```TS
// Actually, I'm using this: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7172b31badc3df45f4386400606dbe24437e3d5a/types/gl-matrix/index.d.ts#L49, but this is enough to trigger the error
class Vec2 extends Float32Array {
    public constructor() {
        super(2);
    }
}

type Vec2orArray = Vec2 | [number, number];
```

**Expected Result**

No linter error since unions are explicitly allowed.

**Actual Result**

Error: `Unhandled in union types are not allowed.eslint(@typescript-eslint/no-type-alias)`

**Additional Info**

Apparently, this hits a case not handled in the rule implementation, see https://github.com/typescript-eslint/typescript-eslint/blob/6be0d811fd7d28fe96a8632da4cd4e81cb517617/packages/eslint-plugin/src/rules/no-type-alias.ts#L243

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.12.0` |
| `@typescript-eslint/parser`        | `1.12.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `12.6.0` |
| `npm`                              | `6.9.0` |
